v5/v5_with_library/v5_stable.ino

```diff
@@ -0,0 +1,2262 @@
+    }
+    updatedContent += line + "\n";
+  }
+
+  // Close the file
+  file.close();
+
+  Println(7, "Updated content : " + updatedContent);
+
+  // Reopen the file for writing, which will erase the previous content
+  file = SPIFFS.open(CONFIG_FILE, "w");
+  if (!file) {
+    println("Failed to open file for writing");
+    return;
+  }
+
+  if (!valueReplaced) {
+    println("Variable not found in file, creating new variable");
+    updatedContent += ("\n" + variableName + ": " + newValue);
+  }
+
+  // Write the modified content back to the file
+  file.print(updatedContent);
+  file.close();
+  Println(7, "Variable {" + variableName + "} has been updated.");
+}
+
+bool hasPackage() {
+  int expiryDate = 0, expiryMonth = 0;
+  if (RTC.date != 0) {
+    if (package_expiry_date == 0)
+      package_expiry_date = getFileVariableValue("packageExpiryDate").toInt();
+    Println(7, "Package expiry date fetched from file: " +
+                   String(package_expiry_date));
+    if (package_expiry_date <= 0) {
+      Println(7, "Package expiry date not initialized in file yet, trying to "
+                 "fetch from ThingSpeak");
+      package_expiry_date = getThingSpeakFieldData(TS_PACKAGE_EXPIRY_DATE);
+      if (package_expiry_date <= 10000) {
+        Println(7,
+                "\t!!!--> Package expiry date is not defined on thingSpeak too"
+                "<---!!!");
+        return false;
+      }
+    }
+  } else {
+    if (updateTime())
+      hasPackage();
+    else
+      alert("Unable to update time from orange pi #1912");
+    Println(7, "Set Time First!");
+  }
+  setField_MonthAndDate(package_expiry_date, expiryMonth, expiryDate);
+  if (RTC.month <= expiryMonth && RTC.month != 0) {
+    Println(7,
+            "\n\n\t $$$ 1:Package is valid RTC value : " + String(RTC.month) +
+                " and expiryMonth value : " + String(expiryMonth));
+    return true;
+  } else if (RTC.month == expiryMonth && RTC.date <= expiryDate &&
+             RTC.month != 0) {
+    Println(7,
+            "\n\n\t $$$ 2: Package is valid RTC value : " + String(RTC.date) +
+                " and expiryDate value : " + String(expiryDate));
+    return true;
+  } else
+    return false;
+}
+
+void setField_MonthAndDate(int &field, int &month, int &date) {
+  if (field > 0 && month == 0 && date == 0) {
+    month = (field / 100) % 100;
+    date = field % 100;
+    Println(7, "From field " + String(field) + " => Month : " + String(month) +
+                   " Date : " + String(date));
+  } else if (field == 0 && month != 0 && date != 0) {
+    field = 10000 + month * 100 + date;
+    Println(7, "From Month : " + String(month) + " & Date : " + String(date) +
+                   " => field : " + String(field));
+  } else {
+    Println(7, "\tError in setField_MonthAndDate function!");
+    Println(7, "Recived data => field: " + String(field) +
+                   " month: " + String(month) + " date: " + String(date));
+    error_codes += "1855";
```

Comment:
Can you please add additional debugging process so at least we got details what scenario is causing this issue and don't forget to mention millis()

Reply:
ok, I'm adding in next commit.

Reply:
resolved in [latest ](https://github.com/mhamidjamil/TTGO_TCall/pull/142/commits/8bf2d7996fb062ebeeaace0efe26d2b5a968047a)code.